app/valve_test/rev1/main.c

```diff
@@ -0,0 +1,320 @@
+
+
+/*------------------------------------------------------------------------------
+ Project Includes
+------------------------------------------------------------------------------*/
+
+/* Application Layer */
+#include "sdr_pin_defines_L0005.h"
+#include "main.h"
+#include "init.h"
+
+/* Module Layer */
+#include "commands.h"
+#include "led.h"
+#include "solenoid.h"
+#include "sensor.h"
+#include "usb.h"
+#include "valve.h"
+
+
+/*------------------------------------------------------------------------------
+ Global Variables
+------------------------------------------------------------------------------*/
+TIM_HandleTypeDef  htim2;  /* Fuel main valve control PWM signals */
+TIM_HandleTypeDef  htim15; /* LOX main valve control PWM signals  */
+UART_HandleTypeDef huart1; /* UART to USB                         */
+UART_HandleTypeDef huart3; /* UART to engine controller           */
+
+
+/*------------------------------------------------------------------------------
+ Application entry point
+------------------------------------------------------------------------------*/
+int main
+	(
+	void
+	)
+{
+/*------------------------------------------------------------------------------
+ Local Variables
+------------------------------------------------------------------------------*/
+uint8_t       command;        /* SDEC command code          */
+USB_STATUS    usb_status;     /* UART/USB status            */
+//VALVE_STATUS  valve_status;   /* Valve module return codes  */
+
+
+/*------------------------------------------------------------------------------
+ Initializations
+------------------------------------------------------------------------------*/
+usb_status    = USB_OK;
+//valve_status  = VALVE_OK;
+
+
+/*------------------------------------------------------------------------------
+ MCU Initialization
+------------------------------------------------------------------------------*/
+HAL_Init           ();   /* CMSIS HAL               */
+SystemClock_Config ();   /* System clock            */
+GPIO_Init          ();   /* GPIO                    */
+USB_UART_Init      ();   /* USB UART                */
+Valve_UART_Init    ();   /* Engine Controller UART  */
+LOX_Valve_TIM_Init ();   /* LOX Valve control timer */
+Fuel_Valve_TIM_Init();   /* Fuel Valve control timer */
+
+
+/*------------------------------------------------------------------------------
+ External Hardware initializations
+------------------------------------------------------------------------------*/
+
+/* Sensor module */
+sensor_init();
+
+/* Indicate successful initialization with green status LED */
+led_set_color( LED_GREEN );
+
+
+/*------------------------------------------------------------------------------
+ Event Loop
+------------------------------------------------------------------------------*/
+while (1)
+	{
+	   for (int i = 0; i < 6; ++i)
+		{
+			solenoid_toggle(i+1);
+		}
+

```

Comment:
Can you use the function solenoid_get_state() here to output the state of each solenoid? Refer to line 324 of solenoid.c.

If all solenoids are toggled, led turns blue